components/Wallet/Replace.tsx

```diff
@@ -0,0 +1,245 @@
+
+  // Calculate maximum transaction fee
+  const maxFee = useMemo<FilecoinNumber | null>(() => {
+    return gasLimit && gasFeeCap ? getMaxGasFee(gasFeeCap, gasLimit) : null
+  }, [gasLimit, gasFeeCap])
+
+  // Set default gas params after loading
+  // the data or disabling expert mode
+  useEffect(() => {
+    if (!expert && gasParams) {
+      setGasPremium(gasParams.gasPremium)
+      setGasLimit(gasParams.gasLimit)
+      setGasFeeCap(gasParams.gasFeeCap)
+    }
+  }, [expert, gasParams])
+
+  // Attempt sending message
+  const onSend = async () => {
+    setIsSending(true)
+    setSendError(null)
+    resetWalletError()
+    const replace = strategy === ReplaceStrategy.CANCEL
+    const newMessage = new Message({
+      to: message.to,
+      from: message.from,
+      nonce: message.nonce,
+      value: replace ? 0 : message.value,
+      method: replace ? 0 : message.method,
+      params: replace ? '' : message.params,
+      gasPremium: gasPremium.toAttoFil(),
+      gasFeeCap: gasFeeCap.toAttoFil(),
+      gasLimit: new BigNumber(gasLimit.toAttoFil()).toNumber()
+    })
+    try {
+      const lotusMessage = newMessage.toLotusType()
+      const signedMessage = await walletProvider.wallet.sign(
+        wallet.address,
+        lotusMessage
+      )
+      const msgValid = await walletProvider.simulateMessage(lotusMessage)
+      if (!msgValid) {
+        throw new Error('Filecoin message invalid. No gas or fees were spent.')
+      }
+      await walletProvider.sendMessage(signedMessage)
+      navigate(router, { pageUrl: PAGE.WALLET_HOME })
+    } catch (e: any) {
+      logger.error(e)
+      setSendError(e)
+    }
+    setIsSending(false)
+  }
+
+  const getTitle = () => {
+    switch (strategy) {
+      case ReplaceStrategy.SPEED_UP:
+        return 'Speed Up Message'
+      case ReplaceStrategy.CANCEL:
+        return 'Cancel Message'
+      default:
+        return ''
+    }
+  }
+
+  const getDescription = () => {
+    if (hasLoadError) return 'Failed to load message information'
+    if (isLoading) return 'Loading message information...'
+    return 'Please confirm the updated message details below'
+  }
+
+  return (
+    <Dialog>
+      <StandardBox>
+        <h2>{getTitle()}</h2>
+        <hr />
+        <p>{getDescription()}</p>
+      </StandardBox>
+      {messageError && (
+        <ErrorBox>Failed to retrieve message: {messageError.message}</ErrorBox>
+      )}
+      {gasParamsError && (
+        <ErrorBox>
+          Failed to calculate gas fees: {gasParamsError.message}
+        </ErrorBox>
+      )}
+      {minGasParamsError && (
+        <ErrorBox>
+          Failed to calculate minimum gas fees: {minGasParamsError.message}
+        </ErrorBox>
+      )}
+      {sendError && (
+        <ErrorBox>Failed to send message: {sendError.message}</ErrorBox>
+      )}
+      {walletError() && (
+        <ErrorBox>The wallet produced an error: {walletError()}</ErrorBox>
+      )}
```

Comment:
I think we should combine into 1 error here and just render a component if error. 

The current setup feels like there could be numerous errors that end up showing both cards (esp with ledger error handling)